src/aleph/vm/controllers/qemu/instance.py

```diff
@@ -0,0 +1,348 @@
+            f"unix:{monitor_socket_path},server,nowait",
+            # Listen for commands on this socket (QMP protocol in json). Supervisor use it to send shutdown or start
+            # command
+            "-qmp",
+            f"unix:{qmp_socket_path},server,nowait",
+            # Tell to put the output to std fd, so we can include them in the log
+            "-serial",
+            "stdio",
+            # Uncomment for debug
+            # "-serial", "telnet:localhost:4321,server,nowait",
+        ]
+        if self.tap_interface:
+            interface_name = self.tap_interface.device_name
+            # script=no, downscript=no tell qemu not to try to set up the network itself
+            args += ["-net", "nic,model=virtio", "-net", f"tap,ifname={interface_name},script=no,downscript=no"]
+
+        cloud_init_drive = await self._create_cloud_init_drive()
+        if cloud_init_drive:
+            args += ["-cdrom", f"{cloud_init_drive.path_on_host}"]
+
+        try:
+            print(*args)
+            self.qemu_process = proc = await asyncio.create_subprocess_exec(
+                *args,
+                stdin=asyncio.subprocess.DEVNULL,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+            logger.debug(f"setup done {self}, {proc}")
+
+            async def handle_termination(proc: Process):
+                await proc.wait()
+                logger.info(f"{self} Process terminated with {proc.returncode} : {str(args)}")
+
+            loop = asyncio.get_running_loop()
+            loop.create_task(handle_termination(proc))
+        except Exception:
+            # Stop the VM and clear network interfaces in case any error prevented the start of the virtual machine.
+            logger.error("VM startup failed, cleaning up network")
+            if self.enable_networking:
+                teardown_nftables_for_vm(self.vm_id)
+            if self.tap_interface:
+                await self.tap_interface.delete()
+            raise
+
+        if self.enable_console:
+            self.process_logs()
+
+        await self.wait_for_init()
+        logger.debug(f"started qemu vm {self} on {self.get_vm_ip()}")
+
+    async def wait_for_init(self) -> None:
+        """Wait for the init process of the instance to be ready."""
+        assert self.enable_networking and self.tap_interface, f"Network not enabled for VM {self.vm_id}"
+
+        ip = self.get_vm_ip()
+        if not ip:
+            msg = "Host IP not available"
+            raise ValueError(msg)
+
+        ip = ip.split("/", 1)[0]
+
+        attempts = 30
+        timeout_seconds = 2.0
+
+        for attempt in range(attempts):
+            try:
+                await ping(ip, packets=1, timeout=timeout_seconds)
+                return
+            except HostNotFoundError:
+                if attempt < (attempts - 1):
+                    continue
+                else:
+                    raise
```

Comment:
Raise what?

Reply:
A bare "raise" in an "Except:" block will re-raise the currently handled exception.